### PR TITLE
Docs: Add guides to plugin overview

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:

Adds the newly written guides to the overview for plugin development.

### CODE REVIEW

File: docs/sources/developers/plugins/_index.md [hunk 1]
```diff
@@ -48,6 +37,25 @@ Ready to learn more? Check out our other tutorials:
...
+
+Improve an existing plugin with one of our guides:
+
+- [Add support for annotations]({{< relref "add-support-for-annotations.md" >}})
+- [Add support for variables]({{< relref "add-support-for-variables.md" >}})
+- [Build a logs data source]({{< relref "build-a-logs-data-source.md" >}})
+
+### Concepts
+
+Deepen your knowledge through a series of high-level overviews of plugin concepts.
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/_index.md:50]:
```suggestion
Deepen your knowledge through a series of high-level overviews of plugin concepts:
```

File: docs/sources/developers/plugins/_index.md [hunk 2]
```diff
@@ -48,6 +37,25 @@ Ready to learn more? Check out our other tutorials:
...
+
+- [Data frames]({{< relref "data-frames.md" >}})
+- [Authentication for data source plugins]({{< relref "authentication.md" >}})
+
+### UI library
+
+Explore the many UI components in our [Grafana UI library](https://developers.grafana.com/ui).
```
Comment (oddlittlebird) [inline docs/sources/developers/plugins/_index.md:57]: Do we want to add a link to the Storybook thing as well?

**Review by oddlittlebird — APPROVED** [review]
Two minor things to address, then good to go.

For the colon, either both list intros need colons or both need periods. I'm just looking for consistency.

File: docs/sources/developers/plugins/_index.md (hunk 2 above)
Comment (marcusolsson) [reply docs/sources/developers/plugins/_index.md:57]: Storybook is the framework used to generate the UI library. It's the same thing. I don't think we should refer to it as "Storybook" in our docs, as it's just the technology we use to generate it.

**Pull request merged**